cmake/configs/base.cmake

```
@@ -400,23 +399,49 @@ endif()
 
 # Setup debug info if enabled.
 if(ENABLE_DEBUG_INFO)
-    if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
-        # Produce full symbolic debugging information.
-        add_compile_options(/Z7)
-        # Ensure a PDB file can be generated for debugging symbols.
-        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /DEBUG")
-    else()
-        # Higher debug levels `-g3`/`-ggdb3` emit additional debug information, including
-        # macro definitions that allow us to evaluate macros such as `p S2C(session)` inside of gdb.
-        # This needs to be in DWARF version 2 format or later - and should be by default - but
-        # we'll specify version 4 here to be safe.
-        add_compile_options(-g3)
-        add_compile_options(-ggdb3)
-        add_compile_options(-gdwarf-4)
-        if("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
-            # Clang requires one additional flag to output macro debug information.
-            add_compile_options(-fdebug-macro)
+    set(BUILD_TYPES_WITH_DEBUG_INFO ${BUILD_MODES})
```

np for consistency: 
```suggestion
    set(BUILD_MODES_WITH_DEBUG_INFO ${BUILD_MODES})
```
or should we rename BUILD_MODES to BUILD_TYPES?